### ISSUE TITLE
Create/Adjust DAO to use between Agent and Server

### ISSUE THREAD

**Issue by benedeki** [AbsaOSS/atum-service#37]
## Background
Data needs to be exchanged between the Agent and the Server

## Feature
Create and update  classes for the data exchange so there's no misalignment between the data sent and received.

## Proposed Solution
1. Probably should be a separate module within the Atum project 

Depends on #39

**Comment by benedeki** [comment]
Needed classes:
* partitioning
* measurement (type of measure)
* checkpoint
* additional data (can be postponed)
* whole Atum context (can be postponed)

**Comment by benedeki** [comment]
This item depends on:
* ~~AbsaOSS/atum-service#39~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.

**Comment by benedeki** [comment]
This item depends on:
* ~~AbsaOSS/atum-service#39~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.
